.travis.yml

```diff
@@ -1,38 +1,106 @@
+conditions: v1
+
 language: python
-python:
-- "3.5"
-- "3.6-dev"
-- "3.7-dev"
-- "nightly"
 
 dist: trusty
 sudo: false
 cache: pip
 
 before_install:
 - pip install --upgrade flit
-install:
-- pushd blurb
-- flit install
-- popd
-script:
-- blurb test
 
 .mixtures:
+- &run-if-tagged
+  if: tag IS present
+- &run-if-cherry-picker
+  if: tag =~ ^cherry\-picker\-v\d+\.\d+\.\d+$
+- &run-if-blurb
+  if: tag =~ ^blurb\-v\d+\.\d+\.\d+$
+- &run-if-cherry-picker-or-untagged
+  if: tag IS NOT present OR tag =~ ^cherry\-picker\-v\d+\.\d+\.\d+$
+- &run-if-blurb-or-untagged
+  if: tag IS NOT present OR tag =~ ^blurb\-v\d+\.\d+\.\d+$
 - &install-and-test-cherry-picker
+  <<: *run-if-cherry-picker-or-untagged
+  env:
+    TARGET_PKG: cherry_picker
   install:
-  - pushd cherry_picker
+  - &cd-to-project pushd "$TARGET_PKG"
   - flit install
-  - popd
   script:
-  - pushd cherry_picker
   - pytest cherry_picker/test.py -v
   - popd
+- &install-and-test-blurb
+  <<: *run-if-blurb-or-untagged
+  env:
+    TARGET_PKG: blurb
+  install:
+  - *cd-to-project
+  - flit install
+  - popd
+  script:
+  - blurb test
+- &deploy-base
+  stage: Publish dists to PYPI
+  <<: *run-if-tagged
+  python: "3.6-dev"
+  install:
+  - *cd-to-project
+  script:
+  - flit build
+  before_deploy:
+  # Add an empty setup.py stub, because pypi provider always calls it
+  - touch setup.py
+  deploy: &deployment-config
+    provider: pypi
+    # `skip-cleanup: true` is required to preserve binary wheel and sdist,
+    # built by during `install` step above.
+    skip-cleanup: true
+    # `skip-existing: true` is required to skip uploading dists, already
+    # present in PYPI instead of failing the whole process.
+    # This happenes when other CI (AppVeyor etc.) has already uploaded
+    # the very same dist (usually sdist).
+    skip-existing: true
+    user: &pypi-user Mariatta  # TODO: consider having a separate "uploader user"
```

Comment:
This has to be core-workflow now.

Reply:
May I expect that you will update this branch yourself?

Reply:
If not - please share the encrypted string and I'll do it myself :)

Reply:
I've updated. At least I think I did it correctly ...

Reply:
Looks fine. Sometimes it doesn't work because of corrupted secret value, but we'll only see this during actual deployment in this repo. Re-encryption usually helps.